test/Microsoft.Data.Analysis.Tests/DataFrame.IOTests.cs

```diff
@@ -1019,5 +1020,187 @@ public void TestMixedDataTypesInCsv()
                 Assert.Equal("", emptyColumn[i]);
             }
         }
+
+        public struct LoadCsvVerifyingHelper
+        {
+            int _columnCount;
+            long _rowCount;
+            string[] _columnNames;
+            Type[] _columnTypes;
+            object[][] _cells;
+
+            public LoadCsvVerifyingHelper(int columnCount, long rowCount, string[] columnNames, Type[] columnTypes, object[][] cells)
+            {
+                _columnCount = columnCount;
+                _rowCount = rowCount;
+                _columnNames = columnNames;
+                _columnTypes = columnTypes;
+                _cells = cells;
+
+            }
+
+            public void VerifyLoadCsv(DataFrame df)
+            {
+                Assert.Equal(_rowCount, df.Rows.Count);
+                Assert.Equal(_columnCount, df.Columns.Count);
+
+                for (int j = 0; j < _columnCount; j++)
+                {
+                    Assert.True(_columnTypes[j] == df.Columns[j].DataType);
+                    Assert.Equal(_columnNames[j], df.Columns[j].Name);
+
+                }
+
+                VerifyColumnTypes(df);
+
+                for (int i = 0; i < _rowCount; i++)
+                {
+                    for (int j = 0; j < _columnCount; j++)
+                    {
+                        Assert.Equal(_cells[i][j], df.Rows[i][j]);
+                    }
```

Comment:
Possibly flatten this Assert a little bit, comparing _cells[i] to df.Rows[i] as a whole instead of individual cells